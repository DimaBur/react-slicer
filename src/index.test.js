import React from 'react'
import renderer from 'react-test-renderer';
import enzyme, { shallow } from 'enzyme'

import Slicer from './index'

import Adapter from 'enzyme-adapter-react-16';

enzyme.configure({ adapter: new Adapter() });

test('Render without crashing', () => {
  const component = renderer.create(
    <Slicer>
      <div key={1}>Item 1</div>
      <div key={2}>Item 2</div>
      <div key={3}>Item 3</div>
      <div key={4}>Item 4</div>
      <div key={5}>Item 5</div>
      <div key={6}>Item 6</div>
      <div key={7}>Item 7</div>
      <div key={8}>Item 8</div>
    </Slicer>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Controls exists', () => {
    const component = shallow(<Slicer>
        <div key={1}>Item 1</div>
        <div key={2}>Item 2</div>
        <div key={3}>Item 3</div>
        <div key={4}>Item 4</div>
        <div key={5}>Item 5</div>
        <div key={6}>Item 6</div>
        <div key={7}>Item 7</div>
        <div key={8}>Item 8</div>
    </Slicer>)

    expect(component.find('.react-slicer__pagination')).toBeTruthy()

})
