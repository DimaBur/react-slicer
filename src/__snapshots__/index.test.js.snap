// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render without crashing 1`] = `
<div
  className="react-slicer"
>
  <div
    className="react-slicer__view"
  >
    <div>
      Item 1
    </div>
    <div>
      Item 2
    </div>
    <div>
      Item 3
    </div>
    <div>
      Item 4
    </div>
  </div>
  <ul
    className="react-slicer__pagination"
  >
    <li
      className="react-slicer__pagination-item react-slicer__pagination-item_prev"
      onClick={[Function]}
    >
      <span
        className="react-slicer__arrow react-slicer__arrow_prev"
      >
        &lt;
      </span>
    </li>
    <li
      className="react-slicer__pagination-item react-slicer__pagination-item_active"
      id={1}
      onClick={[Function]}
    >
      1
    </li>
    <li
      className="react-slicer__pagination-item"
      id={2}
      onClick={[Function]}
    >
      2
    </li>
    <li
      className="react-slicer__pagination-item react-slicer__pagination-item_next"
      onClick={[Function]}
    >
      <span
        className="react-slicer__arrow react-slicer__arrow_next"
      >
        &gt;
      </span>
    </li>
  </ul>
</div>
`;
